decompiler/frontend/binaryninja/handlers/unary.py

```diff
@@ -0,0 +1,93 @@
+            }
+        )
+
+    def lift_unary_operation(self, op_type: OperationType, operation: MediumLevelILOperation, **kwargs) -> UnaryOperation:
+        """Lift the given constant value."""
+        return UnaryOperation(
+            op_type,
+            [self._lifter.lift(x, parent=operation) for x in operation.operands],
+            vartype=self._lifter.lift(operation.expr_type, parent=operation),
+        )
+
+    def lift_cast(self, cast: mediumlevelil.MediumLevelILUnaryBase, **kwargs) -> UnaryOperation:
+        """Lift a cast operation, casting one type to another."""
+        return UnaryOperation(OperationType.cast, [self._lifter.lift(cast.src, parent=cast)], vartype=self._lifter.lift(cast.expr_type))
+
+    def lift_address_of_field(self, operation: mediumlevelil.MediumLevelILAddressOfField, **kwargs) -> Operation:
+        """Lift the address of field operation e.g. &(eax_#1:1)."""
+        if operation.offset == 0:
+            return self.lift_unary_operation(OperationType.address, operation)
+        return BinaryOperation(
+            OperationType.plus,
+            [
+                UnaryOperation(OperationType.address, [operand := self._lifter.lift(operation.src, parent=operation)]),
+                Constant(operation.offset, vartype=operand.type.copy()),
+            ],
+            vartype=self._lifter.lift(operation.expr_type),
+        )
+
+    def _lift_zx_operation(self, instruction: MediumLevelILInstruction, **kwargs) -> UnaryOperation:
+        """Lift zero-extension operation."""
+        inner = self._lifter.lift(instruction.operands[0], parent=instruction)
+        if isinstance(inner.type, Integer) and inner.type.is_signed:
+            unsigned_type = Integer(size=inner.type.size, signed=False)
+            return UnaryOperation(
+                OperationType.cast,
+                [UnaryOperation(OperationType.cast, [inner], unsigned_type)],
+                vartype=self._lifter.lift(instruction.expr_type),
+            )
+        return self.lift_cast(instruction, **kwargs)
+
+    def _lift_load_struct(self, instruction: mediumlevelil.MediumLevelILLoadStruct, **kwargs) -> UnaryOperation:
+        """Lift a MLIL_LOAD_STRUCT_SSA instruction."""
+        return UnaryOperation(
+            OperationType.dereference,
+            [
+                BinaryOperation(
+                    OperationType.plus,
+                    [
+                        UnaryOperation(OperationType.cast, [self._lifter.lift(instruction.src)], vartype=Pointer(Integer.char())),
+                        Constant(instruction.offset),
+                    ],
+                    vartype=self._lifter.lift(instruction.src.expr_type),
+                ),
+            ],
+            vartype=Pointer(self._lifter.lift(instruction.src.expr_type)),
+        )
+
+    def _lift_ftrunc(self, instruction: mediumlevelil.MediumLevelILFtrunc, **kwargs) -> UnaryOperation:
+        """Lift a MLIL_FTRUNC operation."""
```

Comment:
Is truncation semantically equivalent to cast?

Reply:
to the best of my knowledge, the difference is that ftrunc actually returns a floating point value.
e.g. trunc(3.1) -> 3.0 while (int)3.1 -> 3

Reply:
Does it hold for negative numbers also? If yes -> resolved.